docs/specifications/tests/Zone-TP/zone11.md

```diff
@@ -0,0 +1,262 @@
+Z11_SPF1_REDIRECT_AND_ALL       | WARNING |                      | The *Child Zone*’s SPF version 1 policy contains both the "all" keyword and a redirect modifier.
+Z11_SPF1_REDIRECT_NOT_AT_END    | WARNING |                      | The *Child Zone*’s SPF version 1 policy contains a redirect modifier, but it does not appear at the end of the string.
+Z11_SPF1_SYNTAX_ERROR           | ERROR   |                      | The *Child Zone*’s SPF version 1 policy has a syntax error.
+Z11_SPF1_SYNTAX_OK              | INFO    |                      | The *Child Zone*’s SPF version 1 policy has correct syntax.
+Z11_SPF1_TOO_COMPLEX            | ERROR   | count                | Evaluating the *Child Zone*’s SPF version 1 policy would require {count} recursive DNS lookups by an SPF validator. The limit of 10.
+Z11_SPF_NOT_CONFIGURED          | INFO    |                      | The *Child Zone* does not publish an SPF policy.
+Z11_SPF_WRONG_VERSION           | WARNING | spf_version          | The *Child Zone* publishes an SPF policy with invalid version number {spf_version}.
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+## Test procedure
+
+In this section and unless otherwise specified below, the term "[DNS Query]"
+follows the specification for DNS queries as specified in [DNS Query and
+Response Defaults]. The handling of the DNS responses on the DNS queries follow,
+unless otherwise specified below, what is specified for [DNS Response] in the
+same specification.
+
+1. Create a [DNS Query] with query type TXT query and query name *Child Zone*
+   ("TXT query").
+
+2. Create two empty sets of strings, "SPF-Policies" and "SPFv1-Policies".
+
+3. Obtain the set of name server IP addresses using [Method4] and [Method5]
+   ("Name Server IP").
+
+4. For each name server in *Name Server IP* do:
+
+   1. Send *TXT Query* to the name server and collect the [DNS Response].
+
+   2. Go to the next name server IP address if at least one of the following
+      criteria is met:
+
+      1. There is no DNS response.
+      2. [RCODE Name] of the response is not "NoError".
+      3. The AA flag is not set in the response.
+      4. There is no TXT resource record in the response.
+
+   3. If the name server responds with at least one TXT record, then, for each
+      TXT record:
+
+      1. [Concatenate][concatenate] all strings within the TXT record’s data
+         together.
+      2. If the resulting string starts with `v=spf`, then add it to the
+         *SPF-Policies* set.
+
+   4. Go to the next name server.
+
+5. If the *SPF-Policies* set is empty, then output *[Z11_SPF_NOT_CONFIGURED]*
+   and terminate the test.
```

Comment:
I disagree. If a domain name owner does not plan on using their domain name for receiving or sending email, it would make perfect sense for them to specify `v=spf1 -all`.

Reply:
A second question, do you really think it is relevant to output the message for root zone, TLD zones, reverse zones and zones with NULL MX records? See [Zone09].

[Zone09]: https://github.com/zonemaster/zonemaster/blob/develop/docs/specifications/tests/Zone-TP/zone09.md

Reply:
I think it’s useful to point it out either way:

- if a domain owner intends their domain to be used for outgoing e-mail, SPF will protect them and potentially improve their deliverability, so they have something to gain by publishing an SPF policy;
- conversely, if a domain owner does not intend to use their domain name for sending e-mail, they can limit the potential for their domain to be abused by spammers if they publish an SPF policy stating that nobody shall send email from that domain.

In the second case, I’ve read in several places the advice to publish `-all` as an SPF policy, and `p=reject sp=reject` as a DMARC policy, on domains that should not emit mail, as a hardening practice. In both cases there’s something to be gained for a domain owner by implementing SPF.

Reply:
> A second question, do you really think it is relevant to output the message for root zone, TLD zones, reverse zones and zones with NULL MX records? See [Zone09](https://github.com/zonemaster/zonemaster/blob/develop/docs/specifications/tests/Zone-TP/zone09.md).

You are right, we shouldn’t run the test on the root zone, TLD zones (but do we handle edge cases of second-level domains that have the same semantics as TLD zones, such as `co.uk`?).

As for DMARC, its test case is skipped if what [RFC 7489](https://www.rfc-editor.org/rfc/rfc7489) calls the "organizational domain" cannot be deduced from the domain under test.

And we should also exclude from the test case any domain below some of the following special-use domains:

- .arpa;
- .test;
- .localhost;
- .example;
- .invalid.

Reply:
I am not the only one to disagree. There is at least one other testing tool that checks for appropriate SPF and DMARC policies for domains that have null MX records. Which makes sense, because if you clearly state a domain is not to be used to receive email, usually you don’t intend to send e-mail with it either and the right SPF and DMARC policies can help you state that fact.

Here’s an example: https://internet.nl/mail/chargemove.fr/799026/